play-scala-rest-api-example/app/v1/post/PostController.scala

```diff
@@ -27,7 +27,7 @@ class PostController @Inject()(cc: PostControllerComponents)(
       mapping(
         "title" -> nonEmptyText,
         "body" -> text
-      )(PostFormInput.apply)(PostFormInput.unapply)
+      )(PostFormInput.apply)(t => Some(t.title, t.body))
```

Comment:
I don't think this is the correct approach. Rather then using ´(t => Some(t.title, t.body))` you should add an unapply method to `PostFormInput`.
See https://github.com/playframework/playframework/pull/11698/files#diff-3e727831df59776af74d11bfb03065bfe0bb097135126e271e10de55374a90a3 there are also links you should take a look at.